substrate/client/network/transactions_v2/transactions_v2_proposal.md

```diff
@@ -0,0 +1,201 @@
+- The node could retain the bodies of transactions that were broadcasted and then dropped from the pool. This would prevent its reputation from being decreased due to an inability to provide a transaction body when the pool is under high pressure and operating at its limits.
+
+#### [Optional] Transaction Data Low-Fanout (`TxLF`).
+The low-fanout strategy is an optional, easily achievable enhancement aimed at improving transaction propagation latency and network resilience.
+
+The transaction descriptors containing full transaction data are relayed to a small number of (randomly / based on reputation) selected peers. This approach is taken in [etheruem protocol](https://github.com/ethereum/devp2p/blob/master/caps/eth.md#transaction-exchange). When a transaction is submitted via RPC to the local node, the full transaction data may be relayed to an increased number of peers, as no other nodes initially possess this data. This approach helps decrease propagation time by ensuring quicker dissemination.
+
+
+Multiple transaction descriptors shall be batched into a single network notification.
+
+This extension could be used to quickly broadcast transactions from the light nodes to the network.
+
+_Notes_:
+- when a local pool is full the tx can be dropped (e.g. due to lower priority) and there is little we can do about this.
+- `TxLF` only requires extra implementation on the sending side - handling transaction body in `TxDescriptor` is mandatory so no extra support is requried.
+
+
+#### [Optional] Authorities matrix for exchanging transactions.
+Not necessarily a part of protocol itself, could be considered as the optimization of implementation. Authorities could use matrix to effectively exchange txs. Doing so may decrease the latency and improve pools alignment. Similar approach was taken in `polkadot-gossip-support` module ([code](https://github.com/paritytech/polkadot-sdk/blob/98c6ffcea6794d338514cf9bd84446d2f276cb63/polkadot/node/network/gossip-support/src/lib.rs#L786), [doc](https://web.archive.org/web/20221210090830/https://research.web3.foundation/en/latest/polkadot/networking/3-avail-valid.html])). This would require authority-discovery on parachains and probably a more thinking how it could work.
+
+### [Optional] Towards set-reconciliation.
+This section outlines potential future extensions to the transaction protocol. The details of these improvements require further consideration, particularly regarding the cooperation of nodes with different enabled features within the same network. Also some details needs to be figure out - this section only provides a potential directions of future work.
+
+#### [Optional] 32-bit identifier.
+During the handshake peers shall exchange salts. Later salts shall be used to generate 32-bits transaction identifier that is only valid for given pair of peers. This allows to reduce the size of data being flooded to the network, and lays out foundation for introducing a set sketch based reconciliation.
```

Comment:
I guess we need to read about pinsketch to clarify how these 32-bits identifiers (also known as 32-bit fingerprint, right?) are helping reducing the flooded data. Might be worth chatting about it briefly during txpool sync.

Reply:
32bits vs 32bytes per transaction.

Reply:
Ah, right! Thanks!